trace-mini-agent/src/env_verifier.rs

```diff
@@ -178,20 +199,113 @@ async fn get_gcp_metadata_from_body(body: hyper::Body) -> anyhow::Result<GCPMeta
     Ok(gcp_metadata)
 }
 
+async fn verify_azure_environment_or_exit(os: &str) {
+    let now = Instant::now();
+    match ensure_azure_function_environment(Box::new(AzureVerificationClientWrapper {}), os).await {
+        Ok(_) => {
+            debug!("Successfully verified Azure Function Environment.");
+        }
+        Err(e) => {
+            error!("The Mini Agent can only be run in Google Cloud Functions & Azure Functions. Verification has failed, shutting down now. Error: {e}");
+            #[cfg(not(test))]
+            process::exit(1);
+        }
+    }
+    debug!(
+        "Time taken to verify Azure Functions env: {} ms",
+        now.elapsed().as_millis()
+    );
+}
+
+/// AzureVerificationClient trait is used so we can mock the azure function local url response in unit tests
+trait AzureVerificationClient {
+    fn get_function_root_files(&self, path: &Path) -> anyhow::Result<Vec<String>>;
+}
+struct AzureVerificationClientWrapper {}
+
+impl AzureVerificationClient for AzureVerificationClientWrapper {
+    fn get_function_root_files(&self, path: &Path) -> anyhow::Result<Vec<String>> {
+        let mut file_names: Vec<String> = Vec::new();
+
+        let entries = fs::read_dir(path)?;
+        for entry in entries {
+            let entry = entry.map_err(|e| anyhow::anyhow!(e))?;
+            let entry_name = entry.file_name();
+            if entry_name == "node_modules" {
+                continue;
+            }
+
+            file_names.push(entry_name.to_string_lossy().to_string());
+
+            if entry.file_type()?.is_dir() {
```

Comment:
the `function.json` file is located in a sub directory of the main function root folder. This directory is named after the function's handler method which is not exposed via any environment variables. Hence the need to include all files of directories 1 layer deep from the function root dir.